src/mantle/database/model/relations/class-has-one-or-many.php

```diff
@@ -155,34 +156,35 @@ public function add_eager_constraints( Collection $models ): void {
 	 * @param Model[]|Model $model Model instance to save.
 	 * @return Model
 	 */
-	public function save( $model ): Model {
+	public function save( array|Model $model ): Model {
+		if ( is_array( $model ) ) {
+			// Return the first model if saving many.
+			return collect( $this->save_many( $model ) )->first();
+		}
+
 		// Save the model if it doesn't exist.
-		if ( ! is_array( $model ) && ! $model->exists && $model instanceof Updatable ) {
+		if ( ! $model->exists && $model instanceof Updatable ) {
+			$model->save();
+		} elseif ( ! $model->exists && $model instanceof Updatable ) {
```

Comment:
this `if` and `elseif` check the same condition
`! $model->exists && $model instanceof Updatable`